uportal-war/src/main/java/org/jasig/portal/json/rendering/JsonStructureAttributeSource.java

```diff
@@ -21,24 +21,53 @@
 
 import javax.servlet.http.HttpServletRequest;
 
+import org.apache.commons.lang3.StringUtils;
 import org.jasig.portal.layout.IStylesheetUserPreferencesService.PreferencesScope;
 import org.jasig.portal.layout.om.IStylesheetDescriptor;
 import org.jasig.portal.rendering.StylesheetAttributeSource;
 
 /**
  * Returns structure stylesheet descriptor and preferences data
+ *
+ * {@link StylesheetAttributeSource} specific to stylesheets used to produce layout data in JSON format.  This class originally hardcoded the stylesheet name as "DLMMobileColumns" 
+ * and was used to support the "/uPortal/layout.json" call (used for mobile layout rendering).  This class has been updated to support new layout JSON endpoints that will be 
+ * versioned.  For the newer endpoints, this class is now able to use any stylesheet specified.  This class now looks for a request attribute that specifies the stylesheet.  The 
+ * stylesheet will typically be specified in the controller class for the endpoint.  For backwards compatibility, if no stylesheet request attribute is found, the default 
+ * stylesheet used will be the original "DLMMobileColumns".
+ * 
+ * Note that the only difference in the rendering pipeline between the original endpoint and the new versioned endpoints are the stylesheets used.  Because of the number of 
+ * beans required to setup the pipeline (due to the deep chaining of {@link PipelineComponents}), it is desireable to reuse the Spring Framework configuration that sets this 
+ * up rather than duplicating it for each endpoint.  Using a request attribute to specify the stylesheet (instead of using a configured value) allows us to do this 
+ * as each endpoint class or method can specify a different stylesheet.
```

Comment:
Comment width too large (172+).  Please reformat to 100 per future Google code guidelines (https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.4-column-limit).